react-components/src/components/SceneContainer/SceneContainer.tsx

```diff
@@ -0,0 +1,306 @@
+          new THREE.SphereGeometry(9000000, 0, 0),
+          new THREE.MeshBasicMaterial({
+            side: THREE.BackSide,
+            map: new THREE.TextureLoader().load(skyboxUrl)
+          })
+        );
+        viewer.addObject3D(skyboxMesh);
+        skyboxRef.current = skyboxMesh;
+      }
+
+      // Camera translation and rotation
+      {
+        const position = new THREE.Vector3(
+          SceneConfigurationProperties.cameraTranslationX,
+          SceneConfigurationProperties.cameraTranslationY,
+          SceneConfigurationProperties.cameraTranslationZ
+        );
+
+        const euler = new THREE.Euler(
+          SceneConfigurationProperties.cameraEulerRotationX,
+          SceneConfigurationProperties.cameraEulerRotationY,
+          SceneConfigurationProperties.cameraEulerRotationZ,
+          'XYZ'
+        );
+
+        const quaternion = new THREE.Quaternion().setFromEuler(euler);
+        const vec = new THREE.Vector3(0, 0, 1).applyQuaternion(quaternion);
+
+        // There is a bug with setCameraState rotation, so must first set transform
+        // and then lookAt the correct direction
+        viewer.cameraManager.setCameraState({ position });
+        viewer.cameraManager.getCamera().lookAt(position.clone().add(vec));
+      }
+
+      if (scene.items.sceneModels.length > 0) {
+        const sceneModels = scene.items.sceneModels;
+        void Promise.all(
+          sceneModels.map(async (sceneModel) => {
+            const sceneModelProperties = extractProperties(
+              sceneModel.properties
+            ) as SceneModelsProperties;
+
+            const modelId: number = Number(sceneModel.endNode.externalId);
+            const revisionId: number = sceneModelProperties.revisionId;
+
+            if (isNaN(modelId)) {
+              throw new Error('Model id is not a number');
+            }
+            const addModelOptions: AddModelOptions = {
+              modelId,
+              revisionId
+            };
+
+            const transform = new THREE.Matrix4();
+            const scale = new THREE.Matrix4().makeScale(
+              sceneModelProperties.scaleX,
+              sceneModelProperties.scaleY,
+              sceneModelProperties.scaleZ
+            );
+            const euler = new THREE.Euler(
+              sceneModelProperties.eulerRotationX,
+              sceneModelProperties.eulerRotationY,
+              sceneModelProperties.eulerRotationZ,
+              'XYZ'
+            );
+            const rotation = new THREE.Matrix4().makeRotationFromEuler(euler);
+            // Create translation matrix
+            const translation = new THREE.Matrix4().makeTranslation(
+              sceneModelProperties.translationX,
+              sceneModelProperties.translationY,
+              sceneModelProperties.translationZ
+            );
+
+            // Combine transformations
+            transform.multiply(scale).multiply(rotation).multiply(translation);
```

Comment:
`Matrix4.compose()` can be used instead of matrix multiplication. See - https://threejs.org/docs/index.html?q=matrix#api/en/math/Matrix4.compose

Reply:
Fixed in [a776417](https://github.com/cognitedata/reveal/pull/3971/commits/a7764174cfa3137ad95db870c332f72631158749)